hadoop-ozone/dev-support/checks/build.sh

```diff
@@ -26,6 +26,21 @@ else
   MAVEN_OPTIONS="${MAVEN_OPTIONS} -Djacoco.skip"
 fi
 
+PROJECT_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+VERSION_NUMBER=$(echo "${PROJECT_VERSION}"| cut -f1 -d'-')
+EXPECTED_ROCKS_NATIVE_VERSION=${VERSION_NUMBER}".${NATIVE_ROCKS_SHA}"${PROJECT_VERSION:${#VERSION_NUMBER}}
```

Comment:
This will be a strange version string when run locally:

```
1.4.0.-SNAPSHOT
```

because `${NATIVE_ROCKS_SHA}` is only defined in Github Actions.

It would be better to skip the change in Github Actions (`selective_ci_checks.sh` and `ci.yml`), and let `build.sh` be self-contained.  In other words, move SHA calculation here.  Better yet, extract this logic to a helper script, which can be tested in isolation (without a full build).

Also, please make this optional, using an environment variable as condition.

Reply:
done